networkx/classes/tests/test_graph.py

```diff
@@ -29,6 +29,17 @@ def test_nodes(self):
         assert sorted(G.nodes()) == self.k3nodes
         assert sorted(G.nodes(data=True)) == [(0, {}), (1, {}), (2, {})]
 
+    def test_none_node(self):
```

Comment:
It might be worth breaking this test out and parametrizing it over each of the graph classes just to be thorough and ensure that each is completely covered. 

Then again, I don't completely understand the organization of the test suite for the base classes (this may be inherited elsewhere?) so maybe this is already covered.

Reply:
All the tests of the base classes get inherited by the subclasses. So this test is run against each class using `self.Graph` to construct the appropriate graph class object for the test.

This provides two side effects that you might notice when you dive into the tests: 
1) many of the test methods must be rewritten for the subclass because e.g. DiGraph is different from Graph.
2) Be careful when you import the base class testing object... if pytest finds it in the namespace it will run it again -- even though it was already run in the base classes testing module. So, we try to munge the name so pytest doesn't find it in the subclass module.  I'm not sure if we've been entirely successful in removing the extra tests that used to exist -- but its pretty good.